lambda/translprim.ml

```diff
@@ -380,13 +380,15 @@ let lookup_primitive loc ~poly_mode ~poly_sort pos p =
        {[ external id : float# -> float# = "%opaque" ]}
 
        We don't allow non-value layouts for most primitives. This is checked by
-       [prim_has_valid_reprs] in [typing/primitive.ml].
-
-       We don't extract the argument layouts just because it is not needed by
-       the middle-end. *)
+       [prim_has_valid_reprs] in [typing/primitive.ml]. *)
     let (_, repr) = lambda_prim.prim_native_repr_res in
     Lambda.layout_of_extern_repr repr
   in
+  let arg_layouts =
+    List.map
+      (fun (_, repr) -> Lambda.layout_of_extern_repr repr)
+      lambda_prim.prim_native_repr_args
+  in
```

Comment:
It feels slightly wasteful to build this list when only one element of it is used and in only one case (if I understand correctly).  Consider moving the computation of the needed layout to the case where it's used.